coil-base/src/main/java/coil/fetch/ResourceUriFetcher.kt

```diff
@@ -28,7 +28,11 @@ internal class ResourceUriFetcher(
         val resId = data.pathSegments.lastOrNull()?.toIntOrNull() ?: throwInvalidUriException(data)
 
         val context = options.context
-        val resources = context.packageManager.getResourcesForApplication(packageName)
+        val resources = if (packageName == context.packageName) {
```

Comment:
In general, it's because the [`Configuration`](https://developer.android.com/reference/android/content/res/Configuration) is different between:
```
applicationContext.getResources().getConfiguration()
AND
activityContext.getResources().getConfiguration()
```
The `Application` context is first constructed with a default `Configuration` of the system, then `Activity` inherits and extends that configuration with its state, for example, Application context may have `Configuration#orientation = PORTRAIT`, but Activity context has  `Configuration#orientation = LANDSCAPE` therefore the resource retrieved will be different if it's varied by orientation.

`PackageManager#getResourcesForApplication` will returns a `Resource` object with default system `Configuration`. 
So we better use the `context` from `ImageRequest` object.

Reply:
Ah great point! So we're actually using the application resource object here? Sounds like this could potentially fix: https://github.com/coil-kt/coil/issues/954

Reply:
Yeah, it fixes both coil-kt#752 and coil-kt#954